src/ImageSharp/Formats/Gif/GifDecoderCore.cs

```diff
@@ -554,7 +558,7 @@ private void ReadFrameColors<TPixel>(ref Image<TPixel> image, ref ImageFrame<TPi
             {
                 for (int x = descriptorLeft; x < descriptorRight && x < imageWidth; x++)
                 {
-                    int index = Numerics.Clamp(Unsafe.Add(ref indicesRowRef, x - descriptorLeft), 0, colorTableMaxIdx);
+                    int index = Numerics.Clamp(Unsafe.Add(ref indicesRowRef, x - descriptorLeft), 0, Math.Max(transIndex, colorTableMaxIdx));
```

Comment:
Some images use a transparent index that is the same as the length of the palette.

Reply:
As far as I've seen, most viewers treat OOB transparent index as 0.  That's also the behavior written into the PNG spec.  Here's an example where 0 works and last index doesn't.

![bad_transparent_idx](https://user-images.githubusercontent.com/7929102/200686024-51ef470d-6e38-4d24-896b-386b55aae2bf.gif)

Reply:
That's interesting. I found that the second gif in the referenced issue had a frame with a local table that contained 64 colors with a transparent index of 64. Browsers/Windows were rendering it just fine, but we were not.

<img alt="image" src="https://user-images.githubusercontent.com/385879/200696692-6604d670-0cb9-4fe9-9c15-9b37f726d47a.png">

Here's your input gif processed with this PR code.

![OptionalExtensionsShouldBeHandledProperly_Rgba32_issue_2288_3](https://user-images.githubusercontent.com/385879/200695829-e03ca6b8-97b3-47fb-b9ab-ea9b15dcab4f.gif)

Reply:
I use this. Stumbled upon it a few years back and it's proven incredibly useful

https://github.com/ata4/gifiddle

Reply:
Oh wow, that's great, thanks!

I see what's happening here now.  Before this fix, you're not setting a transparent color because it's out of range, but you're clamping the image values into palette range, so pixels meant to be transparent are taking the last color.  This was the result.

![oob_color_is](https://user-images.githubusercontent.com/7929102/200711482-549ed7b1-5a79-4ccc-a385-314212d1ad12.gif)

Now you're treating the last palette color as transparent and clamping OOB image values to that color, making the transparency work, but also making any pixels that were supposed to be that last palette color transparent as well.  It's not creating artifacts as visible as skipping transparency, but it's also not completely correct.

In my implementation, I made the same mistake, but in the other direction.  I use shared code for all animated image formats, so I followed the PNG rules of treating OOB image values as the *first* color in the palette.  That meant I also had to treat invalid transparency index as meaning the first color was supposed to be transparent to get things to look right.  Hence my initial comment 😅

So yeah, the browser behavior is better.  Don't clamp the values but rather assume the palette is actually bigger than the image says.

Reply:
Pushed a better fix now. I treat anything greater than the max palette index or equal to the flagged transparent index as transparent.

I only clamp following that check to ensure we don't hit OOB when reading the palette.